### PR TITLE
Polynomial writer trait

### PR DESCRIPTION
**Pull request by Recmo**
- [ ] Tag the PR with `wip` while in development.
- [ ] Assign yourself as to the PR
- [ ] Assign relevant labels such as `bug`, `enhancement`.
- [ ] Request reviews if the PR is large, complex or you would like an extra pair of eyes to go over it.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] Add new entries to the `Changelog.md`.
- [ ] Update version numbers as needed.

Semver: https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md

https://semver.org/

### CODE REVIEW

File: crypto/stark/examples/pedersen_merkle/inputs.rs [hunk 1]
```diff
@@ -13,39 +13,43 @@ pub struct Claim {
...
+}
 
-    pub fn verify(&self, witness: &Witness) {
-        assert_eq!(self, &Claim::from_leaf_witness(self.leaf.clone(), witness));
+impl From<&Claim> for Vec<u8> {
+    fn from(claim: &Claim) -> Self {
+        let mut bytes: Self = vec![];
+        bytes.extend_from_slice(&claim.path_length.to_be_bytes());
+        bytes.extend_from_slice(&claim.root.as_montgomery().to_bytes_be());
+        bytes.extend_from_slice(&claim.leaf.as_montgomery().to_bytes_be());
```
Comment (pvienhage) [inline crypto/stark/examples/pedersen_merkle/inputs.rs:51]: The path: .as_montgomery().to_bytes_be() is used quite a bit throughout our code, and I think it would read better as a path like .as_montgomery_bytes(). That change is out of scope for this pr though.

File: crypto/stark/examples/pedersen_merkle/component.rs [hunk 2]
```diff
@@ -50,11 +49,13 @@ impl Component for MerkleTreeLayer {
     type Claim = ();
     type Witness = (FieldElement, FieldElement, bool);
 
+    fn claim(&self, _witness: &Self::Witness) -> Self::Claim {}
+
     fn dimensions(&self) -> (usize, usize) {
```
Comment (Recmo) [inline crypto/stark/examples/pedersen_merkle/component.rs:54]: Split into `num_polynomials` and `polynomial_size`.

File: crypto/stark/src/component2/mod.rs [hunk 3]
```diff
@@ -11,50 +12,85 @@ use crate::{
...
 
 pub use empty::Empty;
 pub use fold::Fold;
 pub use horizontal::Horizontal;
+pub use mapped::Mapped;
 pub use test::Test;
 pub use vertical::Vertical;
 
+/// A set of Polynomials represented by their values at roots of unity.
+pub trait PolyWriter {
```
Comment (Recmo) [inline crypto/stark/src/component2/mod.rs:25]: Rename `PolynomialWriter`
